packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -183,7 +189,7 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
       isExactActive: link?.isExactActive ?? computed(() => to.value === router.currentRoute.value.path),
       route: link?.route ?? computed(() => router.resolve(to.value)),
       async navigate () {
-        await navigateTo(href.value, { replace: props.replace, external: props.external })
+        await navigateTo(href.value, { replace: props.replace, external: isExternal.value || hasTarget.value })
       },
     } satisfies ReturnType<typeof useLink> & {
       to: ComputedRef<string | RouteLocationRaw>
```

Comment:
I apologize for not noticing that this is also present here. 

This revision did not change this part. Would it be strange to modify them together?

```suggestion
      to: ComputedRef<RouteLocationRaw>
```